sphinxcontrib/needs/directives/need.py

```diff
@@ -81,15 +81,13 @@ def run(self):
 
         # ToDo: Keep this in directive!!!
         collapse = self.options.get("collapse", None)
-        if isinstance(collapse, str) and len(collapse) > 0:
+        if isinstance(collapse, str):
             if collapse.upper() in ["TRUE", 1, "YES"]:
                 collapse = True
             elif collapse.upper() in ["FALSE", 0, "NO"]:
                 collapse = False
             else:
                 raise Exception("collapse attribute must be true or false")
-        else:
-            collapse = getattr(env.app.config, "needs_collapse_details", None)
```

Comment:
Why can this be removed?
It's load the value from conf.py if it is not set via directive option?

Reply:
it states in the comments and documentation that "needs_collapse_details" is deprecated, in favour of setting in `layout`. It's also not set anywhere as a configuration option in the `setup` function, so `collapse = getattr(env.app.config, "needs_collapse_details", None)` will *always* return None